contracts/VotingMachines/Governance.sol

```diff
@@ -0,0 +1,509 @@
+        proposal.state = ProposalState.NoneBoosted;
+        // solium-disable-next-line security/no-block-members
+        proposal.submittedTime = now;
+        proposals[proposalId] = proposal;
+        NewProposal(proposalId, _numOfChoices, msg.sender, _paramsHash);
+        return proposalId;
+    }
+
+  /**
+   * @dev Cancel a proposal, only the owner can call this function and only if allowOwner flag is true.
+   * @param _proposalId the proposal ID
+   */
+    function cancelProposal(bytes32 _proposalId) public onlyProposalOwner(_proposalId) votable(_proposalId) returns(bool) {
+        //This is not allowed.
+        return false;
+    }
+
+    /**
+     * @dev staking function
+     * @param _proposalId id of the proposal
+     * @param _vote a value between 0 to and the proposal number of choices.
+     * @param _amount the betting amount
+     * @return bool true - the proposal has been executed
+     *              false - otherwise.
+     */
+    function stake(bytes32 _proposalId, uint _vote, uint _amount) public returns(bool) {
+        if (execute(_proposalId)) {
+            return true;
+        }
+        if (proposals[_proposalId].state != ProposalState.NoneBoosted) {
+            return false;
+        }
+        Proposal storage proposal = proposals[_proposalId];
+        uint amount = _amount;
+      // Check valid vote:
+        require(_vote <= proposal.numOfChoices);
+        bytes32 paramsHash = getParametersFromController(Avatar(proposals[_proposalId].avatar));
+        Parameters memory orgParams = parameters[paramsHash];
+        assert(amount > orgParams.minimumStaketingFee);
+        stakingToken.transferFrom(msg.sender, address(this), amount);
+
+        proposal.stakes[_vote] = amount.add(proposal.stakes[_vote]);
+        proposal.totalStakes = amount.add(proposal.totalStakes);
+        proposal.stakers[msg.sender] = Staker({
+            amount: amount,
+            vote: _vote
+        });
+      // Event:
+        Stake(_proposalId, msg.sender, _vote, _amount);
+      // execute the proposal if this vote was decisive:
+        return execute(_proposalId);
+    }
+
+  /**
+   * @dev voting function
+   * @param _proposalId id of the proposal
+   * @param _vote a value between 0 to and the proposal number of choices.
+   * @return bool true - the proposal has been executed
+   *              false - otherwise.
+   */
+    function vote(bytes32 _proposalId, uint _vote) public votable(_proposalId) returns(bool) {
+        return internalVote(_proposalId, msg.sender, _vote, 0);
+    }
+
+  /**
+   * @dev voting function with owner functionality (can vote on behalf of someone else)
+   * @param _proposalId id of the proposal
+   * @return bool true - the proposal has been executed
+   *              false - otherwise.
+   */
+    function ownerVote(bytes32 _proposalId, uint , address ) public onlyProposalOwner(_proposalId) votable(_proposalId) returns(bool) {
+      //This is not allowed.
+        return false;
```

Comment:
Should this revert instead of returning false, or perhaps add an `allowed(false)` modifier?

Reply:
ownerVote is not allowed for this votingMachine ( yet).
It is there just because of the voting interface.
Not sure it should revert.
One should not call that and if it does it will just do nothing.

Reply:
Reverting (or maybe better, using a modifier like allowed(false)) would instantly and unequivocally apprise the user that they have done something wrong.

Returning false, unless they are checking the return value, would be silent, thus potentially delaying the time until which the user would realize they've done something wrong.

Reply:
I guess it is a good practice to check return value for each call.
revert for this case seems too heavy tool.